workspaces/tech-radar/plugins/tech-radar-common/src/schema.ts

```diff
@@ -0,0 +1,148 @@
+  id: z.string(),
+  // Display name of the Ring
+  name: z.string(),
+  // Color used for entries in particular Ring, Supports any value parseable by {@link https://www.npmjs.com/package/color-string | color-string}
+  color: z.string(),
+  // Description of the Ring
+  description: z.string().optional(),
+});
+
+/**
+ * Tech Radar Ring which indicates stage of {@link RadarEntry}
+ *
+ * @public
+ */
+export type RadarRing = z.infer<typeof RadarRingParser>;
+
+/**
+ * Parser for {@link RadarQuadrant}
+ *
+ * @public
+ */
+export const RadarQuadrantParser = z.object({
+  // ID of the Quadrant
+  id: z.string(),
+  // Display name of the Quadrant
+  name: z.string(),
+});
+
+/**
+ * Tech Radar Quadrant which represent area/topic of {@link RadarEntry}
+ *
+ * @public
+ */
+export type RadarQuadrant = z.infer<typeof RadarQuadrantParser>;
+
+const RadarEntryLinkParser = z.object({
+  // URL of the link
+  url: z.string(),
+  // Display name of the link
+  title: z.string(),
+});
+
+/**
+ * Indicates how {@link RadarEntry} moved though {@link RadarRing} on the {@link RadarEntry} timeline
+ *
+ * @public
+ */
+export enum MovedState {
+  /**
+   * Moved down
+   */
+  Down = -1,
+  /**
+   * Didn't move
+   */
+  NoChange = 0,
+  /**
+   * Move up
+   */
+  Up = 1,
+}
+
+const RadarEntrySnapshotParser = z.object({
+  // Point in time when change happened
+  date: z.coerce.date(),
+  // ID of {@link RadarRing}
+  ringId: z.string(),
+  // Description of change
+  description: z.string().optional(),
+  // Indicates trend compared to previous snapshot
+  moved: z.nativeEnum(MovedState).optional(),
+});
+
+/**
+ * Parser for {@link RadarEntry}
+ *
+ * @public
+ */
+export const RadarEntryParser = z.object({
+  // React key to use for this Entry
+  key: z.string(),
+  // ID of this Radar Entry
+  id: z.string(),
+  // ID of {@link RadarQuadrant} this Entry belongs to
+  quadrant: z.string(),
+  // Display name of the Entry
+  title: z.string(),
+  // User-clickable URL when rendered in Radar
+  url: z.string().optional(),
+  // History of the Entry moving through {@link RadarRing}
+  timeline: z.array(RadarEntrySnapshotParser),
+  // Description of the Entry
+  description: z.string().optional(),
+  // User-clickable links to provide more information about the Entry
+  links: z.array(RadarEntryLinkParser).optional(),
+});
+
+/**
+ * Single Entry in Tech Radar
+ *
+ * @public
+ */
+export type RadarEntry = z.infer<typeof RadarEntryParser>;
+
+/**
+ * Parser for the response from Tech Radar loader.
+ *
+ * @public
+ */
+export const TechRadarLoaderResponseParser = z.object({
+  quadrants: z.array(RadarQuadrantParser),
+  rings: z.array(RadarRingParser),
+  entries: z.array(RadarEntryParser),
+});
+
+/**
+ * Response from that contains all the data for a Tech Radar graph;
+ *  {@link RadarQuadrant}, {@link RadarRing}, and {@link RadarEntry}.
+ *
+ * @public
+ */
+export type TechRadarLoaderResponse = z.infer<
+  typeof TechRadarLoaderResponseParser
+>;
```

Comment:
I think it's better to copy-paste the types as they were defined in the frontend instead of inferring them from zod, so that the api-report will be cleaner. The main reason is that if people use a different version of zod than the version defined in tech-radar-common, they might get typings issues.
One way to keep the typescript types and the zod validator in sync can be to define the validator like this:

```ts
const techRadarLoaderResponseParser: z.ZodSchema<TechRadarLoaderResponse> = z.object({
   ...stuff
});
```

with the approach above, typescript screams if the validator doesn't match the defined type.

```suggestion
export interface TechRadarLoaderResponse {
  /**
   * Quadrant of Tech Radar. Should be 4
   */
  quadrants: RadarQuadrant[];
  /**
   * Rings of Tech Radar
   */
  rings: RadarRing[];
  /**
   * Entries visualised in Tech Radar
   */
  entries: RadarEntry[];
}
```